### PR TITLE
Monitor our backend (web) server and deployed sessions

### PR DESCRIPTION
**Pull request by amolenaar**
Add monitoring for the backend.

The idea is that the backend can provide metrics regarding running sessions, and the session (deployment) state.

* `backend_database_sessions` (gauge) shows the number of sessions registered in our backend database
* `backend_deployed_sessions{state}` (gauge) shows the number of pods deployed in the sessions namespace, with their state
* `backend_sessions_started` (counter) total number of sessions started since application startup
* `backend_sessions_killed` (counter) total number of sessions terminated, either by user or timeout
* `starlette_*` HTTP request metrics: requests, responses, request times, exceptions, requests in progress

### CODE REVIEW

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]

File: backend/capellacollab/__main__.py [hunk 1]
```diff
@@ -63,9 +66,17 @@ async def schedule_termination_of_idle_sessions():
...
+    capellacollab.sessions.metrics.register()
+
+
 app = FastAPI(
     title="Capella Collaboration",
-    on_startup=[startup, schedule_termination_of_idle_sessions],
+    on_startup=[
+        startup,
+        schedule_termination_of_idle_sessions,
+        register_metrics,
```
[the remark this answers is not in the document]
Comment (amolenaar) [reply backend/capellacollab/__main__.py:78]: True, this function was probably a left-over from some other approaches I tried.

File: backend/capellacollab/sessions/metrics.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+class DeployedSessionsCollector:
+    def collect(self) -> t.Iterable[prometheus_client.core.Metric]:
+        metric = prometheus_client.core.GaugeMetricFamily(
+            "backend_deployed_sessions",
+            "Sessions running in the sessions namespace",
+            labels=("phase",),
+        )
+        operator = operators.get_operator()
+        pods = operator.get_pods(label_selector=None)
```
[the remark this answers is not in the document]
Comment (amolenaar) [reply backend/capellacollab/sessions/metrics.py:33]: This query is already reading all pods (in the sessions namespace), right?

[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/capellacollab/sessions/metrics.py:33]: `operator.get_pods` returns all pods in a namespace, not filtering for session Pods. The metrics endpoint for active sessions should just return session pods and no other pods.

File: backend/capellacollab/__main__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MoritzWeber0) [reply backend/capellacollab/__main__.py:78]: Do you plan to remote it then?

File: backend/capellacollab/sessions/metrics.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (amolenaar) [reply backend/capellacollab/sessions/metrics.py:33]: image

The metric now has a `workload` label, that tells us what kind of workload it is (session, job, cronjob)

**Review by MoritzWeber0 — APPROVED** [review]

**Pull request merged**